### PR TITLE
test: Add tests for contexts interface

### LINKED ISSUE #343: Add integration tests for contexts interface
We serialize and transform context data from Godot types to underlying native types. This can lead to some surprises in the process, and needs to be tested. We could use #340 for this.

### PR DESCRIPTION
**Pull request by limbonaut**
This PR adds tests for contexts interface with detailed JSON validation. This will help detect regressions and inconsistencies between platform implementations. Extracted from #326.
- Closes #343

### CODE REVIEW

File: project/test/suites/test_contexts_json.gd [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		.must_contain("/contexts/os") \
+		.verify()
+
+	assert_json(json).describe("Test Godot contexts presence") \
+		.at("/") \
+		.must_contain("/contexts/godot_engine") \
+		.must_contain("/contexts/display") \
+		.must_contain("/contexts/environment") \
+		.must_contain("/contexts/godot_performance") \
+		.verify()
```
[the remark this answers is not in the document]
Comment (limbonaut) [reply project/test/suites/test_contexts_json.gd:199]: I believe it's more readable this way.

File: project/test/suites/test_contexts_json.gd [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+	var json: String = await capture_event_and_get_json(SentrySDK.create_event())
+
+	assert_json(json).describe("Verify edge values in context data") \
+		.is_object() \
+		.at("/contexts/edge_data") \
+		.must_contain("empty_string", "") \
+		.must_contain("zero_value", 0) \
+		.must_contain("false_value", false) \
+		.verify()
```
[the remark this answers is not in the document]
Comment (limbonaut) [reply project/test/suites/test_contexts_json.gd:136]: Fixed in 6bddc95

File: project/test/suites/test_contexts_json.gd [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+
+func test_empty_context_is_captured() -> void:
+	SentrySDK.set_context("empty_context", {})
+
+	var json: String = await capture_event_and_get_json(SentrySDK.create_event())
+
+	assert_json(json).describe("Verify empty context is present") \
+		.at("/contexts/") \
```
[the remark this answers is not in the document]
Comment (limbonaut) [reply project/test/suites/test_contexts_json.gd:62]: No it's fine.
